src/components/shared/Image.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react'
+import { useRef, useEffect, forwardRef, useImperativeHandle } from 'react'
+import { useAsync } from 'react-use'
+import Services from '../../extension/service'
+import { Skeleton, SkeletonProps } from '@material-ui/lab'
+export interface ImageProps {
+    children?: never
+    src: string | Blob
+    /**
+     * Where should the request be sent?
+     * `current` means the origin of the current page (which means might be blocked by CSP policy)
+     *
+     * `extension` means the request will be sent in the `chrome-extension://` protocol. You need to have the permission to the target site or the site has configured CORS policy correctly.
+     *
+     * `auto` means this component will try to decide this based on "src" property
+     *
+     * @default auto
+     */
+    origin?: 'current' | 'extension' | 'auto'
+    /**
+     * Which tag should be used to render the image?
+     *
+     * - `img` is ignored when `src` is ArrayBuffer.
+     *
+     * - `img` is ignored when `origin` is `extension`.
+     * @default auto
+     */
+    component?: 'img' | 'canvas'
+    width: number
+    height: number
+    loading?: boolean
+    // usability
+    canvasProps?: React.DetailedHTMLProps<React.CanvasHTMLAttributes<HTMLCanvasElement>, HTMLCanvasElement>
+    imgProps?: React.DetailedHTMLProps<React.ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement>
+    SkeletonProps?: SkeletonProps
+    className?: string
```

Comment:
Remove the `className` property. The user can do the same thing with `canvasProps` or `imgProps`. BTW, better use `CanvasProps` and `ImgProps`.

Reply:
it's provided for ergonomic reason

Reply:
It's better to design like MUI components. It will be easier for someone else who uses this component. I still advise using a big camel style as the properties' name.

Reply:
there are no inner DOM structures so `*Props` is enough (no need to use `classes`)

Reply:
> BTW, better use `CanvasProps` and `ImgProps`.

The naming is also following MUI style cause `img` and `canvas` are native elements meanwhile `Skeleton` are React component